lib/pos_epoch_complete_hook.go

```diff
@@ -163,6 +162,43 @@ func (bav *UtxoView) runEpochCompleteEpochRollover(epochNumber uint64, blockHeig
 		FinalBlockHeight:                nextEpochFinalBlockHeight,
 		CreatedAtBlockTimestampNanoSecs: blockTimestampNanoSecs,
 	}
+	return nextEpochEntry, nil
+}
+
+func (bav *UtxoView) computePrevEpochEntry(epochNumber uint64, blockHeight uint64, view uint64, blockTimestampNanoSecs uint64) (*EpochEntry, error) {
+	if epochNumber == 0 {
+		return nil, errors.New("computePrevEpochEntry: epochNumber is 0, this should never happen")
+	}
+	snapshotAtEpochNumber, err := bav.ComputeSnapshotEpochNumberForEpoch(epochNumber - 1)
+	if err != nil {
+		return nil, errors.Wrapf(err, "computePrevEpochEntry: problem computing snapshot epoch number: ")
+	}
+	snapshotGlobalParamsentry, err := bav.GetSnapshotGlobalParamsEntryByEpochNumber(snapshotAtEpochNumber)
+	if err != nil {
+		return nil, errors.Wrapf(err, "computePrevEpochEntry: problem retrieving snapshot global params entry: ")
+	}
+
+	// Calculate the PrevEpoch's InitialBlockHeight.
+	prevEpochInitialBlockHeight, err := SafeUint64().Sub(blockHeight, snapshotGlobalParamsentry.EpochDurationNumBlocks+1)
+	if err != nil {
+		return nil, errors.Wrapf(err, "computePrevEpochEntry: problem calculating PrevEpochInitialBlockHeight: ")
+	}
+	prevEpochEntry := &EpochEntry{
+		EpochNumber:                     epochNumber - 1,
+		InitialBlockHeight:              prevEpochInitialBlockHeight,
+		InitialView:                     view - 1, // Note that this value is incorrect and can't be computed.
```

Comment:
Maybe zero this out too since we know it will always be wrong
```suggestion
		InitialView: 0
```